### PR TITLE
New gtsummary

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* These updates should make the package function with the coming gtsummary v2.0 release.

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [ ] Has `NEWS.md` been updated with the changes from this pull request under the heading "`# gtreg (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Has the version number been incremented using `usethis::use_version(which = "dev")` 
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -29,8 +29,8 @@ jobs:
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}
           - {os: ubuntu-latest,   r: 'oldrel-2'}
-          - {os: ubuntu-latest,   r: 'oldrel-3'}
-          - {os: ubuntu-latest,   r: 'oldrel-4'}
+          # - {os: ubuntu-latest,   r: 'oldrel-3'}
```
Comment (ddsjoberg) [inline .github/workflows/R-CMD-check.yaml:32]: gsummary is using the base R pipe and the `|> foo(x = _)`, so it doesn't work on 4 previous releases anymore. But we'll get back to it soon enough :)

File: tests/testthat/_snaps/style_xxx.md [hunk 2]
```diff
@@ -1,1571 +1,92 @@
 # style_xxx works with tbl_ae family
 
     Code
-      as_gt(t1, id = "t1_001")
```
Comment (ddsjoberg) [inline tests/testthat/_snaps/style_xxx.md:4]: It turns out the raw HTML from gt built tables changes slightly pretty often. I changed this to a data frame. But we are no longer testing the headers FYI

File: tests/testthat/test-tbl_ae.R [hunk 3]
```diff
@@ -263,7 +263,7 @@ test_that("tbl_ae() works", {
   )
   expect_equal(
     as_tibble(tbl, col_labels = FALSE)$stat_1,
-    c("2.0 / 3.0 (66.67%)", "2.0 / 3.0 (66.67%)", "0.0 / 3.0 (0.00%)")
+    c("2.0 / 3.0 (66.67%)", "2.0 / 3.0 (66.67%)", "0.0 / 3.0 (0%)")
```
Comment (ddsjoberg) [inline tests/testthat/test-tbl_ae.R:266]: I checked this and the change I made here is also consistent with the last version of the package. I don't know why was previously `0.00%`.

**Review by shannonpileggi — APPROVED** [review]
you are the best!! all of this looks great, thank you so much.

one thing i'd like to better understand & discuss in a f2f is `tbl_summary` vs `tbl_reg_summary` and `inheritParams`. Also, themes & `tbl_reg_summary`

**Comment by ddsjoberg** [comment]
> you are the best!! all of this looks great, thank you so much.
> 
> one thing i'd like to better understand & discuss in a f2f is `tbl_summary` vs `tbl_reg_summary` and `inheritParams`. Also, themes & `tbl_reg_summary`

We'll have a lil powwow about it in seattle ;)

**Pull request merged**